packages/api/es/collections.js

```diff
@@ -29,20 +30,38 @@ class Collection extends BaseSearch {
           },
         },
         aggs: {
-          hashes: {
-            terms: {
-              field: '_uid',
-              size: ES_MAX_AGG,
+          collections: {
+            filter: {
+              term: {
+                _type: 'collection',
+              },
             },
             aggs: {
-              stats: {
-                children: {
-                  type: 'granule',
+              name: {
+                terms: {
+                  field: 'name',
+                  size: ES_MAX_AGG,
```

Comment:
Why are we re-adding the size delimiter? I guess we've established that it didn't cause the bug, but setting it to its absolute maximum value gives me some concern? 

Is it because we want the stats to be accurate? I thought we were just going to un-break them and not worry about the accuracy for now

Reply:
I commented a bit more on this in the Jira ticket, but just a bit.  

The issue here is that bucket aggregations really aren't the right way to do a query like this and get all the records - the bucket returns a 'top hits' which can be size limited.  

That's particularly relevant for the stats values, as getting 'some' of the statuses means that the counts are going to be inaccurate in ways that aren't predictable for each collection if there are more than the default (10?) for any of the term aggregations. 

 This approach is consistent with what's in master currently, whereas I did not include it in the backport (see review on #1884).   The term buckets this is being applied to are name/version and the stats coming back from the granule child query (which should already hit *all granules* for each name/version combination.         

Unless I'm missing something (entirely possible!) this might fall down is if there was an extremely large number of collections/versions, or a dramatic increase in the stats tracked for the granules.

We should be using (per the docs for 6+ they explicitly suggest this if you're trying to get a query that has all possible keys) https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-composite-aggregation.html, but that requires a newer elastic search version and all that implies. 

If we feel we need to setup a testing apparatus (e.g. dump a bunch of data into ES) for load testing on this , I'm more than game.

Reply:
So are you saying that this is the only way to ensure that we're getting all granule stats for a collection given the type of query we're using? And that there's no "better" type of query for us to use given our version of ES?

To me, just using a hardcoded value which is literally the maximum possible value feels like we're asking for trouble. The PRs coming to make the stats query run for just a single collection won't really help this situation, will it?

Reply:
> So are you saying that this is the only way to ensure that we're getting all granule stats for a collection given the type of query we're using?

More accurately I believe it's the only way to ensure we get stats for all collections in our query.